aiidalab_widgets_base/elns.py

```diff
@@ -0,0 +1,361 @@
+            description="Optimize atomic positions and/or unit cell employing Quantum ESPRESSO. Quantum ESPRESSO is preferable for small structures with no cell dimensions larger than 15 Å. Additionally, you can choose to compute electronic properties of the material such as band structure and density of states.",
+        ).to_html_string()
+
+        geo_opt.value += AppIcon(
+            icon="https://raw.githubusercontent.com/lsmo-epfl/aiidalab-epfl-lsmo/develop/miscellaneous/logos/LSMO.png",
+            link=f"https://aiidalab-demo.materialscloud.org/user-redirect/apps/apps/aiidalab-lsmo/multistage_geo_opt_ddec.ipynb?structure_uuid={self.node.uuid}",
+            description="Optimize atomic positions and unit cell with CP2K. CP2K is very efficient for large and/or porous structures. A structure is considered large when any cell dimension is larger than 15 Å. Additionally, you can choose to assign point charges to the atoms using DDEC.",
+        ).to_html_string()
+
+        return geo_opt
+
+    def get_isotherm_tab(self):
+        isotherm = ipw.HTML()
+        isotherm.value += AppIcon(
+            icon="https://raw.githubusercontent.com/lsmo-epfl/aiidalab-epfl-lsmo/develop/miscellaneous/logos/LSMO.png",
+            link=f"https://aiidalab-demo.materialscloud.org/user-redirect/apps/apps/aiidalab-lsmo/compute_isotherm.ipynb?structure_uuid={self.node.uuid}",
+            description="Compute adsorption isotherm of the selected material using the RASPA code. Typically, one needs to optimize geometry and compute the charges of material before isotherm. However, if this is already done, you can go for it.",
+        ).to_html_string()
+
+        return isotherm
+
+    def get_geometry_analysis_tab(self):
+        geo_analysis = ipw.HTML()
+        geo_analysis.value += AppIcon(
+            icon="https://raw.githubusercontent.com/lsmo-epfl/aiidalab-epfl-lsmo/develop/miscellaneous/logos/LSMO.png",
+            link=f"https://aiidalab-demo.materialscloud.org/user-redirect/apps/apps/aiidalab-lsmo/pore_analysis.ipynb?structure_uuid={self.node.uuid}",
+            description="Perform geometry analysis of a material employing Zeo++ code.",
+        ).to_html_string()
+
+        return geo_analysis
+
+
+class ElnImportWidget(ipw.VBox):
+    node = traitlets.Instance(Node, allow_none=True)
+
+    def __init__(self, **kwargs):
+        # Used to output additional settings.
+        self._output = ipw.Output()
+
+        # Communicate to the user if something isn't right.
+        error_message = ipw.HTML()
+        super().__init__(children=[error_message], **kwargs)
+
+        eln = connect_to_eln(**kwargs)
+
+        if eln is None:
+            error_message.value = f"""Warning! The access to ELN {kwargs['eln_name']} is not configured. Please follow <a href="https://aiidalab-demo.materialscloud.org/user/aliaksandr.yakutovich@epfl.ch/apps/apps/aiidalab-widgets-base/eln_configure.ipynb" target="_blank">the link</a> to configure it."""
```

Comment:
we should not hardcode the username, but i cannot make a suggestion as I do not know how to inspect what the username is :D

Reply:
fixed.